spec/support/anno_auth_helper.rb

```diff
@@ -0,0 +1,18 @@
+require 'webmock/rspec'
+
+module AnnoAuthHelper
+    def stub_anno_auth(group_id, has_access, server)
+        body = if group_id.nil? && has_access.nil? 
+                 [{}]
+               else
+                 [{"group_id": group_id, "has_access": has_access}]
+               end
+        stub_request(:get, server).
+          to_return(status: 200, body: body.to_json)   
+    end
+
+    def set_anno_auth_token(group_id, has_access, server, auth_header)
```

Comment:
set_anno_auth_token is doing two very different things here: stubbing a web server and setting the request header. It would be better if stub_anno_auth and set_anno_auth_token be called separately, rather than one calling the other.

Reply:
I put them together so that you didnt have to make two calls in every spec but still had the ability to override stub_anno_auth when needed (for example in the 403 tests).

Reply:
Ignore my last comment @gigamorph. You are right, it does look better. Will update.

Reply:
I don't think we lose much in having an additional line. I think it is clearer to readers to separate `stub_anno_auth` and `set_anno_auth_token`. If you really want to combine methods, `set_anno_auth_token` and the following call to`post_to_create` are a lot more homogeneous to each other, because they are both about sending a request. Again, I don't think you need try too hard to combine those either.